### PR TITLE
Issue#3 themes

### PR DESCRIPTION
**Pull request by rdmueller**
This PR adds support for themes

The plugin now
- loads the dark theme when darcula is active in intellij or the kennedy theme when another theme is active in intellij
- this not only works when you start the IDE, but also when the theme is changed and then a new drawio file is opened
- there s still a bug with the currently open and visible file when you change the theme. Workaround is to reload this file

### CODE REVIEW

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt [hunk 1]
```diff
@@ -105,6 +103,13 @@ abstract class BaseDrawioWebView(val lifetime: Lifetime) {
 
     private var requestId = 0
 
+    public fun reload(uiTheme: String) {
+        this.uiTheme = uiTheme
+        //initializeSchemeHandler(uiTheme)
+        this.panel.browser.cefBrowser.reload()
+        send(OutgoingMessage.Event.Load("",1))
```
Comment (hediet) [inline src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt:110]: What is `Load` doing?

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt [hunk 2]
```diff
@@ -105,6 +103,13 @@ abstract class BaseDrawioWebView(val lifetime: Lifetime) {
 
     private var requestId = 0
 
+    public fun reload(uiTheme: String) {
+        this.uiTheme = uiTheme
+        //initializeSchemeHandler(uiTheme)
```
Comment (hediet) [inline src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt:108]: Why is this out commented?

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt [hunk 3]
```diff
@@ -54,9 +72,22 @@ class DiagramsEditor(private val project: Project, private val file: VirtualFile
...
-    private fun saveFile(data : ByteArray) {
+    @Override
+    override public fun globalSchemeChange(scheme: EditorColorsScheme?) {
+        System.out.println("test");
+        if (scheme?.name=="Darcula" || scheme?.name=="High contrast") {
+            uiTheme = "dark"
+
+        } else {
+            uiTheme = "kennedy"
+        }
```
Comment (hediet) [inline src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt:86]: This is duplicated code!
Maybe you can use `PropertyView<T>` to abstract this!

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt [hunk 4]
```diff
@@ -1,30 +1,48 @@
...
-import java.util.*
 import javax.swing.JComponent
+import javax.swing.UIManager
 
 
-class DiagramsEditor(private val project: Project, private val file: VirtualFile) : FileEditor {
+class DiagramsEditor(private val project: Project, private val file: VirtualFile) : FileEditor, EditorColorsListener {
     private val lifetimeDef = LifetimeDefinition()
     private val lifetime = lifetimeDef.lifetime
+    private var uiTheme = "dark" // "dark" or "kennedy"
```
Comment (hediet) [inline src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt:23]: Can you make this an observable property? See how the `xml` field in the drawio instance is implemented!

**Review by hediet — CHANGES_REQUESTED** [review]
Looks good, but I think the code can be improved a bit ;)

**Comment by hediet** [comment]
I think it is not super beautiful to re-register the scheme handler... Maybe this can be solved with `index.html?theme=...` + js that checks the query parameter?

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt (hunk 1 above)
Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt:110]: this is part of the WIP - doesn't work yet. 
The `reload()` results in having only the menu. I think there is a load event missing at this point.

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt (hunk 2 above)
Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt:108]: also part of the WIP. I though this would help but it didn't

Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt]: I was wrong - it is helpful. re-added it.

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt (hunk 1 above)
Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt]: removed it and replaced it with a todo comment

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt (hunk 3 above)
Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt]: I guess I will learn during my Kotlin training how to do this :-)

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt (hunk 4 above)
Comment (rdmueller) [reply src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/DiagramsEditor.kt:23]: I though about an enumeration but at the moment, I just want to get it to work...

**Comment by rdmueller** [comment]
yes, this could be a solution, but the reload of the page seems not to trigger the load event. That seems to be the main problem here.

**Pull request merged**